### DISCUSSION TITLE
mandatory fields

### DISCUSSION THREAD

**Discussion by zpj61** [fattureincloud/api#87] [Q&A] [answered]
Hi, we are testing invoice creation and would be useful to have a list of which fields  are mandatory and witch not, for example if "rivalsa" is inserted (4%) the field "amount_rivalsa" is calculated automatically or must be specified? Currency, language  are they mandatory? Which fields are setted automatically?

**Comment by emajo** [comment, marked as answer]
Hi @zpj61 the mandatory fields are marked with a red `*` in our doc, however it's impossible to document all the conditional mandatory fields, we suggest to make some tests and see if the results are what's expected.
In the case of the `rivalsa` and many other fields as `cassa` you can either set the percentage in `rivalsa` or the `amount_rivalsa`, if you set the percentage the amount will be calculated by the api.

Currency and language are not mandatory as they are by default € and italian.

Feel free to contact us if you have more doubts,
thanks,
Emanuele.
